core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -807,8 +801,8 @@ private[spark] class BlockManager(
         // Now that the block is in either the memory or disk store,
         // tell the master about it.
         info.size = size
-        if (tellMaster) {
-          reportBlockStatus(blockId, info, putBlockStatus)
+        if (tellMaster && info.tellMaster) {
```

Comment:
Could we rename the flag to `removedByMaster` or similar? Might also consider renaming `tellMaster` to `trackedByMaster` but that's probably a bigger change.